### PR TITLE
swarm_behaviors: 1.2.0-1 in 'kinetic/distribution.yaml' [bloom]

### PR DESCRIPTION
**Pull request by tropappar**
Increasing version of package(s) in repository `swarm_behaviors` to `1.2.0-1`:

- upstream repository: https://github.com/cpswarm/swarm_behaviors.git
- release repository: https://github.com/cpswarm/swarm_behaviors-release.git
- distro file: `kinetic/distribution.yaml`
- bloom version: `0.9.0`
- previous version for package: `null`

## position

```
* Added: Obstacle detection
* Changed: Refactor library structure
* Changed: Split up navigation library, create position library and move parts to abstraction library
* Changed: Move obstacle avoidance to abstraction library
* Changed: Always use local coordinates
* Fixed: Force C++11
* Fixed: Goal computation
* Contributors: Micha Sende, Omar Morando
```

## swarm_behaviors

```
* First public release
* Added: Package position
* Added: Package velocity
* Added: Package uav_optimal_coverage
* Added: Package ugv_random_walk
* Changed: Create stack for swarm_behaviors library
* Changed: Refactor library structure
* Fixed: Force C++11
* Changed behavior algorithms: Targets handled by swarm functions library
* Changed behavior algorithms: Check if movement was successful
* Changed coverage algorithms: Return state aborted once coverage finishes
* Changed random coverage algorithms: Read RNG seed from parameter, use random seed otherwise
* Fixed behavior algorithms: Correctly handle state
* Fixed coverage algorithms: Returning of target
* Changed: Split up navigation library, create position library and move parts to abstraction library
* Added to position: Obstacle detection
* Changed position: Move obstacle avoidance to abstraction library
* Changed position: Always use local coordinates
* Fixed position: Goal computation
* Changed velocity: Limit velocity for small distances
* Changed uav_optimal_coverage: Use position setpoint
* Changed uav_optimal_coverage: Use current position to determine next waypoint
* Fixed uav_optimal_coverage: Stop moving when behavior finishes
* Fixed uav_optimal_coverage: Completely traverse path
* Changed uav_random_direction: UAVs depart in different directions
* Fixed uav_random_direction: Change direction when obstacle detected
* Fixed uav_simple_tracking: Handling of target rescued event
* Contributors: Micha Sende, Omar Morando
```

## uav_local_coverage

```
* Changed: Refactor library structure
* Changed: Return state aborted once coverage finishes
* Changed: Check if movement was successful
* Changed: Targets handled by swarm functions library
* Fixed: Correctly handle state of behavior algorithm
* Fixed: Force C++11
* Fixed: Returning of target
* Contributors: Micha Sende
```

## uav_optimal_coverage

```
* Changed: Refactor library structure
* Changed: Use position setpoint
* Changed: Return state aborted once coverage finishes
* Changed: Use current position to determine next waypoint
* Changed: Targets handled by swarm functions library
* Fixed: Correctly handle state of behavior algorithm
* Fixed: Stop moving when behavior finishes
* Fixed: Completely traverse path
* Fixed: Force C++11
* Fixed: Returning of target
* Initial release of uav_optimal_coverage
* Contributors: Micha Sende
```

## uav_random_direction

```
* Changed: Refactor library structure
* Changed: UAVs depart in different directions
* Changed: Read RNG seed from parameter, use random seed otherwise
* Changed: Return state aborted once coverage finishes
* Changed: Check if movement was successful
* Changed: Targets handled by swarm functions library
* Fixed: Change direction when obstacle detected
* Fixed: Correctly handle state of behavior algorithm
* Fixed: Force C++11
* Fixed: Returning of target
* Contributors: Micha Sende
```

## uav_simple_tracking

```
* Fixed: Handling of target rescued event
* Changed: Refactor library structure
* Changed: Check if movement was successful
* Changed: Targets handled by swarm functions library
* Fixed: Correctly handle state of behavior algorithm
* Fixed: Force C++11
* Contributors: Micha Sende
```

## ugv_random_walk

```
* Changed: Refactor library structure
* Changed: Read RNG seed from parameter, use random seed otherwise
* Changed: Return state aborted once coverage finishes
* Changed: Check if movement was successful
* Changed: Targets handled by swarm functions library
* Fixed: Correctly handle state of behavior algorithm
* Fixed: Force C++11
* Fixed: Returning of target
* Initial release of ugv_random_walk
* Contributors: Micha Sende
```

## velocity

```
* Changed: Refactor library structure
* Changed: Limit velocity for small distances
* Fixed: Force C++11
* Initial release of velocity
* Contributors: Micha Sende
```

### CODE REVIEW

File: kinetic/distribution.yaml [hunk 1]
```diff
@@ -14907,6 +14907,31 @@ repositories:
...
       version: master
     status: developed
+  swarm_behaviors:
+    doc:
+      type: git
+      url: https://github.com/cpswarm/swarm_behaviors.git
+      version: kinetic-devel
+    release:
+      packages:
+      - position
```
Comment (tfoote) [inline kinetic/distribution.yaml]: This package does not meet our naming requirements. It's way too generic.

File: kinetic/distribution.yaml [hunk 2]
```diff
@@ -14907,6 +14907,31 @@ repositories:
...
+      - swarm_behaviors
+      - uav_local_coverage
+      - uav_optimal_coverage
+      - uav_random_direction
+      - uav_simple_tracking
+      - ugv_random_walk
+      - velocity
```
Comment (tfoote) [inline kinetic/distribution.yaml]: This is also too generic.

**Review by tfoote — CHANGES_REQUESTED** [review]
Overalll this looks good, but there's two packages that too generic names of pure abstract concepts. Prefixing them with `cpswarm_` is probably the right solution.


https://www.ros.org/reps/rep-0144.html

File: kinetic/distribution.yaml (hunk 2 above)
Comment (tropappar) [reply kinetic/distribution.yaml]: What about:

- swarm_behaviors_position
- swarm_behaviors_velocity

**Comment by tropappar** [comment]
What about:
- swarm_behaviors_position
- swarm_behaviors_velocity

**Comment by tfoote** [comment]
Those sounds good.

**Comment by tropappar** [comment]
Requested changes have been applied and moved to a new pull request #23398.

**Pull request closed without merging**